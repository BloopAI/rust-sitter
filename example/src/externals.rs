#[rust_sitter::grammar("externals", "src/externals/scanner.c")]
#[allow(dead_code)]
mod grammar {
    #[rust_sitter::language]
    #[derive(Debug)]
    pub struct Language {
        #[rust_sitter::leaf(pattern = r"\d+", transform = |v| v.parse().unwrap())]
        v: Option<i32>,
        #[rust_sitter::leaf(text = "_")]
        _s: (),
        #[rust_sitter::leaf(text = ".")]
        _d: Option<()>,
    }

    #[rust_sitter::external]
    #[rust_sitter::extra]
    struct WhiteSpaces;
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn externals_grammar() {
        insta::assert_debug_snapshot!(grammar::parse("_"));
        insta::assert_debug_snapshot!(grammar::parse("_."));
        insta::assert_debug_snapshot!(grammar::parse("1_"));
        insta::assert_debug_snapshot!(grammar::parse("23_."));
        insta::assert_debug_snapshot!(grammar::parse(" _  "));
        insta::assert_debug_snapshot!(grammar::parse("_ , ."));
        insta::assert_debug_snapshot!(grammar::parse("1_;"));
        insta::assert_debug_snapshot!(grammar::parse("23;; , ;;_."));
    }
}
