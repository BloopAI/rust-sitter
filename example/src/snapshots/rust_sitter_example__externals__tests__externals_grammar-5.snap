---
source: example/src/externals.rs
assertion_line: 30
expression: "grammar::parse(\" _  \")"
---
Ok(
    Language {
        v: None,
        _s: (),
        _d: None,
    },
)
